crates/subspace-core-primitives/src/lib.rs

```diff
@@ -457,6 +457,46 @@ impl RootBlock {
     }
 }
 
+/// Index of piece on disk
+pub type PieceOffset = u64;
+/// Piece index in consensus
+pub type PieceIndex = u64;
+
+/// Hash of `PieceIndex`
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PieceIndexHash(pub u64);
```

Comment:
Namespace for these hashes is 256-bit just like for node IDs, not u64 (remember, we'll use XOR metric for checking which pieces need to be replaced). I think this abstraction is not needed, just use `Sha256Hash` or at most create a wrapper around that (seems premature though).